### PR TITLE
Add cache file permission config to Enlight_Template_Manager

### PR DESCRIPTION
**Pull request by Crease29**
### 1. Why is this change necessary?

In a setup where the SSH/FTP user isn't the owner of files that were written by the webserver user, it's impossible to delete template cache files because of its fixed chmods.

### 2. What does this change do, exactly?

The following chmod settings from the config will be applied to smarty template cache files:
```
    'cache' => [
        'backendOptions' => [
            'hashed_directory_perm' => 0777 & ~umask(),
            'cache_file_perm'       => 0775 & ~umask(),
        ],
    ],
```

### 3. Describe each step to reproduce the issue or behaviour.

:heavy_minus_sign: 

### 4. Please link to the relevant issues (if any).

:heavy_minus_sign: 

### 5. Which documentation changes (if any) need to be made because of this PR?

:heavy_minus_sign: 

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Library/Enlight/Template/Manager.php [hunk 1]
```diff
@@ -66,13 +66,15 @@ class Enlight_Template_Manager extends Smarty
      */
     public function __construct($options = null)
     {
+        $cacheOptions = Shopware()->Container()->getParameter('shopware.cache');
```
Comment (hlohaus) [inline engine/Library/Enlight/Template/Manager.php]: Dependency Injection would be nice

Comment (Crease29) [reply engine/Library/Enlight/Template/Manager.php]: Okay, I will check this ;)

**Comment by Crease29** [comment]
Done @hlohaus :blush:

**Comment by Crease29** [comment]
I'd like to push this. Isn't this a quick-pick that can easily be merged?

**Comment by soebbing** [comment]
Hi @Crease29, your PR has been merged in 250d670ad0788ad55a3f659c7ea1236777b9511b, thank you for your contribution! 💙

**Pull request closed without merging**